services/job_subscriber.go

```diff
@@ -71,14 +83,16 @@ func (js *JobSubscriber) Disconnect() {
 }
 
 // OnNewHead resumes all pending job runs based on the new head activity.
-func (js *JobSubscriber) OnNewHead(head *models.BlockHeader) {
-	pendingRuns, err := js.Store.JobRunsWithStatus(models.RunStatusPendingConfirmations)
+func (js *jobSubscriber) OnNewHead(head *models.BlockHeader) {
+	pendingRuns, err := js.store.JobRunsWithStatus(models.RunStatusPendingConfirmations, models.RunStatusInProgress)
 	if err != nil {
 		logger.Error(err.Error())
 	}
+
+	ibn := head.ToIndexableBlockNumber()
 	for _, jr := range pendingRuns {
-		if _, err := ExecuteRunAtBlock(jr, js.Store, jr.Result, head.ToIndexableBlockNumber()); err != nil {
-			logger.Error(err.Error())
+		if err := js.store.RunChannel.Send(jr.Result, ibn); err != nil {
```

Comment:
Let's just inline `head.ToIndexableBlockNumber()` into the method call instead of using `ibn`.

Reply:
It's the same for every iteration of the loop, I'd rather just leave it up top.

Reply:
Didn't realize it was outside a loop. Good stuff.